ethstats/state.go

```diff
@@ -129,10 +129,12 @@ func (s *State) WriteBlock(b *Block) error {
 		return err
 	}
 
-	// add the transactions for each block
-	for _, txn := range b.Txs {
-		if _, err := tx.Exec(`INSERT INTO block_transactions (block_hash, txn_hash) VALUES ($1, $2)`, b.Hash, txn.Hash); err != nil {
-			return err
+	if config.ShouldSaveBlockTxs {
+		// add the transactions for each block
+		for _, txn := range b.Txs {
+			if _, err := tx.Exec(`INSERT INTO block_transactions (block_hash, txn_hash) VALUES ($1, $2)`, b.Hash, txn.Hash); err != nil {
```

Comment:
Interesting that it inserts in a loop like this lol. Removing this should greatly reduce the resource requirements for this db.